lua/ulx/modules/sh/cfc_constraint_checker.lua

```diff
@@ -128,42 +128,47 @@ local function compileConstraintResults( _, ply, constraintCounts )
     return blockData
 end
 
+local function getMsgCArgs( constraintData )
+    local args = {}
+    for _, data in pairs( constraintData ) do
+        table.Add( args, getPlyCountsMsgData( data.ply, data.counts ) )
+    end
+
+    return args
+end
 
 function cmd.checkConstraints( caller, targetPlys, showPlysWithNoConstraints )
     local perPlyConstraints = countConstraints( targetPlys )
-    local dataBlocks = {}
 
     ulx.fancyLogAdmin( caller, true, "#A checked the constraints of #T", targetPlys ) -- Alert staff console of the command being used
 
+    -- Convert constraint data to list and remove players with no constraints if necessary
+    local constraintCountsList = {}
     for _, ply in pairs( targetPlys ) do
-        local constraintCounts = perPlyConstraints[ply]
-        if showPlysWithNoConstraints or constraintCounts.Total > 0 then
-            local block = compileConstraintResults( caller, ply, constraintCounts )
-            table.insert( dataBlocks, block )
+        local plyCounts = perPlyConstraints[ply]
+        if showPlysWithNoConstraints or plyCounts.Total > 0 then
+            table.insert( constraintCountsList, {
+                ply = ply,
+                counts = plyCounts
+            } )
         end
     end
+    table.sort( constraintCountsList, function( a, b )
+        return a.counts.Total > b.counts.Total
+    end )
 
-    -- sort the data if needed to
-    if #dataBlocks > 1 then
-        table.sort( dataBlocks, function( a, b )
-            a = string.sub( a[7], 7, #a[7] - 2 )
-            b = string.sub( b[7], 7, #b[7] - 2 )
-            return tonumber( a ) < tonumber( b )
-        end )
-    end
-
-    local finalData = {}
-    for _, block in pairs( dataBlocks ) do table.Add( finalData, block ) end
-
+    -- Create args for MsgC using constraint count list and send to client
+    -- TODO move visualization code clientside
     net.Start( "CFC_ULX_ConstraintResults" )
-    net.WriteTable( finalData )
+        net.WriteTable( getMsgCArgs( constraintCountsList ) )
```

Comment:
I moved everything associated with visualizing the data out of this function and into getMsgCArgs. Keeping that code separate from the rest of the logic makes it easier to work with